pkgs/stdenv/generic/setup.sh

```diff
@@ -12,7 +12,7 @@ fi
 
 shopt -s inherit_errexit
 
-if (( "${NIX_DEBUG:-0}" >= 6 )); then
+if test "${NIX_DEBUG:-0}" -ge 6; then
```

Comment:
```suggestion
if [[ ${NIX_DEBUG:-0} -ge 6 ]]; then
```
We can use Bash expressions here instead of builtin commands.

Reply:
I’m fine with this. Note that `[[ … ]]` isn’t quite the panacea against having to quote things you might expect, though; `[[ $x = *$y ]]` is not the same as `[[ $x = "*$y" ]]`. In some ways I prefer `test` because it doesn’t tempt you to slip on the discipline, although it’s obviously worse in a bunch of other ways. (Did you know that Bash is not a very good language?)

Reply:
Also note that this restores the wonky evaluation behaviour of `(( … ))` (which isn’t so bad, if we check the syntax of `NIX_DEBUG` at the start, but it’s still surprising).

Reply:
> I’m fine with this.

Do you mean with @tie's proposed update, or with the original `test`?

Reply:
I’m fine with anything :) I’m pretty sure `[[ $x -gt 1 ]]` is identical to `(( x >= 1 ))`, so it’s effectively the same as reverting my suggestion, but if we do up‐front validation of `$NIX_DEBUG` then they’re equivalent. I slightly prefer my conservative `test` version for the reasons given, but I wanted to make it clear that I wasn’t strenuously objecting to the effective undo.

Reply:
>`[[ $x = *$y ]]` is not the same as `[[ $x = "*$y" ]]`

Uh, yes, this is a feature (glob matching)? But yeah Bash and “Unix” shells in general have some rea-a-ally weird edge cases that I can’t explain sometimes, e.g. https://github.com/NixOS/nixpkgs/pull/328208#discussion_r1683566765.

That said, I’m usually fine with anything as well, but I prefer the code base to be consistent, and unfortunately Bash (and its syntax) is deep-rooted in Nixpkgs’s stdenv, so we might as well use it to its full potential…

FWIW in most cases `[[ … ]]` is identical to `[ … ]` (i.e. `test`) but is an expression instead of command.

Reply:
Yep, it’s a feature. But did you know that `[[ $x =~ [a-z] ]]` and `[[ $x =~ "[a-z]" ]]` are also different because of the same feature? I mean, you learn these things eventually, but people get told “`[[ … ]]` is just like `[ … ]` except you don’t have to quote stuff”, and it’s actually more like “`[[ … ]]` is just like `[ … ]` except the quoting rules are *even more complicated and weird*”…

Reply:
I've returned to `[[` style tests.